### PR TITLE
Added SampleUpdate call for public SDK

### PR DESCRIPTION
**Pull request by JoseLuisNipo**
New call in public SDK for the "Sample update" feature

### CODE REVIEW

File: Library/Models/ClearSurveySampleModel.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Model that holds the encrypted data and the initialization vector that was used to encrypt it
```
Comment (evadelnuevo) [inline Library/Models/ClearSurveySampleModel.cs]: Is this the correct description?

File: Library/Models/SampleColumnUpdate.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+
+namespace Nfield.Models
+{
+    /// <summary>
+    /// Survey setting.
```
Comment (evadelnuevo) [inline Library/Models/SampleColumnUpdate.cs]: Is this the correct description? (also in ColumnName and Value)

File: Library/Models/SampleFilter.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ namespace Nfield.Models
     /// <summary>
     /// class to define sample filters
     /// </summary>
-    internal class SampleFilter
+    public class SampleFilter
```
Comment (evadelnuevo) [inline Library/Models/SampleFilter.cs]: Is this change intended?

File: Tests/Services/NfieldSurveySampleTests.cs [hunk 4]
```diff
@@ -527,6 +525,74 @@ public void TestClearAsync_ParamsAreOk_Successful()
...
+        {
+            int sampleRecordId = 1;
+            var columns = new List<SampleColumnUpdate>();
+            columns.Add(new SampleColumnUpdate { ColumnName = "dummyCol", Value = "dummyVal" });
+
+            var mockedNfieldConnection = new Mock<INfieldConnectionClient>();
+            var mockedHttpClient = CreateHttpClientMock(mockedNfieldConnection);
+
+            mockedHttpClient.Setup(client => client.PutAsJsonAsync($"{ServiceAddress}Surveys/{SurveyId}/Sample/Update",
+                It.Is<SurveyUpdateSampleRecordModel>(c => string.IsNullOrEmpty(SurveyId) && c.ColumnUpdates.Any(n => n.ColumnName == "dummyCol"))))
```
Comment (evadelnuevo) [inline Tests/Services/NfieldSurveySampleTests.cs]: The SurveyId condition is unnecessary, instead, you should add the sampleRecordId to the condition, as it's used to build the SurveyUpdateSampleRecordModel

**Comment by evadelnuevo** [comment]
Another thing: the PR should be against your ci branch (not master)

File: Library/Models/ClearSurveySampleModel.cs (hunk 1 above)
Comment (JoseLuisNipo) [reply Library/Models/ClearSurveySampleModel.cs]: no - fixed in commit 61b186c5103693627e2922122aa481b00ae36a20

File: Library/Models/SampleColumnUpdate.cs (hunk 2 above)
Comment (JoseLuisNipo) [reply Library/Models/SampleColumnUpdate.cs]: no - fixed in commit 61b186c5103693627e2922122aa481b00ae36a20

File: Library/Models/SampleFilter.cs (hunk 3 above)
Comment (JoseLuisNipo) [reply Library/Models/SampleFilter.cs]: yes, it's necessary

File: Tests/Services/NfieldSurveySampleTests.cs (hunk 4 above)
Comment (JoseLuisNipo) [reply Tests/Services/NfieldSurveySampleTests.cs]: fixed in commit 61b186c5103693627e2922122aa481b00ae36a20

File: Library/Models/SampleFilter.cs (hunk 3 above)
Comment (evadelnuevo) [reply Library/Models/SampleFilter.cs]: 👍
